trunner/harness/unity.py

```diff
@@ -46,13 +48,17 @@ def unity_harness(dut: Dut, ctx: TestContext) -> Optional[TestResult]:
         parsed = dut.match.groupdict()
 
         if idx == 0:
-            if parsed["status"] == "FAIL":
-                last_fail = parsed
+            if parsed["status"] in ["FAIL", "IGNORE"]:
+                last_assertion = parsed
         elif idx in (1, 2):
-            # TODO we do not consider INFO/IGNORE messages
+            # TODO we do not consider INFO messages
             if parsed["status"] == "FAIL":
-                if last_fail["path"] == parsed["path"] and last_fail["line"] == parsed["line"]:
-                    parsed["msg"] = last_fail["msg"]
+                if last_assertion["path"] == parsed["path"] and last_assertion["line"] == parsed["line"]:
+                    parsed["msg"] = last_assertion["msg"]
+            if parsed["status"] == "IGNORE":
+                if last_assertion["status"] == parsed["status"]:
+                    parsed["msg"] = last_assertion["msg"]
+                    last_assertion["status"] = None
```

Comment:
Let's set `last_assertion` to an empty dict before line 63. I think is more readable, it clearly says that we've read the test final message and we clear the last assertion because it should not be re-used later.